paz/models/classification/cnn2Plus1.py

```diff
@@ -0,0 +1,354 @@
+import einops
+import random
+import tensorflow as tf
+from tensorflow.keras import Sequential
+from tensorflow.keras.models import Model
+from tensorflow.keras.layers import (Conv3D, Input, BatchNormalization, Layer, GlobalAveragePooling3D, ReLU, Flatten,
+                                     LayerNormalization, Dense, add)
+from typing import Literal, get_args
+
+Architecture_Options = Literal["CNN2Plus1D", "CNN2Plus1D_Filters", "CNN2Plus1D_Layers", "CNN2Plus1D_Light",
+                               "CNN2Plus1D_18"]
+
+"""
+References:
+    - [A Closer Look at Spatiotemporal Convolutions for Action Recognition](https://arxiv.org/abs/1711.11248v3)
+    - [Video classification with a 3D convolutional neural network](https://www.tensorflow.org/tutorials/video/video_classification#load_and_preprocess_video_data)
+"""
+
+
+class Conv2Plus1D(Layer):
+    def __init__(self, filters, kernel_size, padding):
+        """
+        A sequence of convolutional layers that first apply the convolution operation over the
+        spatial dimensions, and then the temporal dimension.
+        """
+        super().__init__()
+        initializer_glorot_spatial = tf.keras.initializers.GlorotUniform(seed=random.randint(0, 1000000))
+        initializer_glorot_temporal = tf.keras.initializers.GlorotUniform(seed=random.randint(0, 1000000))
+
+        self.seq = Sequential([
+            # Spatial decomposition
+            Conv3D(filters=filters,
+                   kernel_size=(1, kernel_size[1], kernel_size[2]),
+                   padding=padding,
+                   kernel_initializer=initializer_glorot_spatial),
+            # Temporal decomposition
+            Conv3D(filters=filters,
+                   kernel_size=(kernel_size[0], 1, 1),
+                   padding=padding,
+                   kernel_initializer=initializer_glorot_temporal)
+        ])
+
+    def call(self, x):
+        return self.seq(x)
+
+
+class ResidualMain(Layer):
+    """
+    Residual block of the model with convolution, layer normalization, and the
+    activation function, ReLU.
+    """
+
+    def __init__(self, filters, kernel_size):
+        super().__init__()
+        self.seq = Sequential([
+            Conv2Plus1D(filters=filters,
+                        kernel_size=kernel_size,
+                        padding='same'),
+            LayerNormalization(),
+            ReLU(),
+            Conv2Plus1D(filters=filters,
+                        kernel_size=kernel_size,
+                        padding='same'),
+            LayerNormalization()
+        ])
+
+    def call(self, x):
+        return self.seq(x)
+
+
+class Project(Layer):
+    """
+    Project certain dimensions of the tensor as the data is passed through different
+    sized filters and downsampled.
+    """
+
+    def __init__(self, units):
+        super().__init__()
+        initializer_glorot = tf.keras.initializers.GlorotUniform(seed=random.randint(0, 1000000))
```

Comment:
You could simply import this initializer on the top and call it `GlorotUniform` instead of `tf.keras.initializers.GlorotUniform`. It would be easier to read. Moreover I would suggest that you scientific notation i.e. `1e6` or this notation `1_000_000`. It just makes it more clear what values you are using.